src/source/Crypto/Crypto.h

```diff
@@ -8,10 +8,32 @@ extern "C" {
 #endif
 
 #ifdef KVS_USE_OPENSSL
-#define KVS_RSA_F4                  RSA_F4
-#define KVS_MD5_DIGEST_LENGTH       MD5_DIGEST_LENGTH
-#define KVS_SHA1_DIGEST_LENGTH      SHA_DIGEST_LENGTH
+#define KVS_RSA_F4             RSA_F4
+#define KVS_MD5_DIGEST_LENGTH  MD5_DIGEST_LENGTH
+#define KVS_SHA1_DIGEST_LENGTH SHA_DIGEST_LENGTH
+
+// https://www.openssl.org/docs/man3.1/man3/MD5.html
+#if (OPENSSL_VERSION_NUMBER >= 0x30000000L)
+#define KVS_MD5_DIGEST(m, mlen, ob)                                                                                                                  \
```

Comment:
Not sure why the formatting looks like this. But locally it is fine and clang-format passed locally and on CI.